protocol/daemons/types/health_checkable.go

```diff
@@ -0,0 +1,11 @@
+package types
+
+import "context"
+
+// HealthCheckable is a common interface for services that can be health checked.
+type HealthCheckable interface {
+	// HealthCheck returns an error if a service is unhealthy. This method takes
+	// a context to allow the caller to use a context-based timeout with the call.
+	// If the service is healthy, this method returns nil.
+	HealthCheck(ctx context.Context) (err error)
```

Comment:
What should be the behavior if the daemon is in STOPPED state? Or is this not a concern because `HealthCheck` method won't be able to get called at all?

Reply:
In the stopped state, the last successful update timestamp will never be refreshed and the daemon will become unhealthy after a waiting period of 5 min.